hivemind/moe/client/beam_search.py

```diff
@@ -151,11 +147,14 @@ async def _get_initial_beam(
                 maybe_prefix_data = await pending_task
                 if maybe_prefix_data is not None and isinstance(maybe_prefix_data.value, dict):
                     successors = {
-                        coord: UidEndpoint(uid=match.value[0], peer_info=PeerInfo.from_tuple(match.value[1]))
+                        coord: ExpertInfo(uid=match.value[0], peer_id=PeerID.from_base58(match.value[1]))
                         for coord, match in maybe_prefix_data.value.items()
                         if isinstance(coord, Coordinate)
-                        and isinstance(getattr(match, "value", None), list)
+                        and isinstance(match, ValueWithExpiration)
+                        and isinstance(match.value, tuple)
                         and len(match.value) == 2
+                        and is_valid_uid(match.value[0])
+                        and isinstance(match.value[1], str)
```

Comment:
Maybe extract this 5-line validation into a function?

Reply:
done